src/app/components/shared/Table/components/TableHeader/Sorter.tsx

```diff
@@ -50,28 +56,28 @@ const StyledLabel = styled.div<LabelProps>`
   @keyframes rotate-back {
     0% { transform: rotate(0deg); }
     100% { transform: rotate(-180deg); }
-  }   
+  }
 `
 
 const StyledIcon = styled(Icon)<IconProps>`
   margin: -${ themeSpacing(0.5) } ${ themeSpacing(1) } 0 ${ themeSpacing(2) };
   visibility: ${ ({ isSelected }) => isSelected ? "visible" : "hidden" };
-  color: ${ themeColor("tint", "level6") };            
+  color: ${ themeColor("tint", "level6") };
 `
 
 const Sorter: React.FC<Props> = ({ header, index, sorting, onChangeSorting }) => {
-  const isSelected = sorting.columnKey === index
+  const isSelected = sorting?.index === index
   // Only show arrow down icon when order is "DESCEND" and sorting is already selected.
-  const iconType = isSelected && sorting.order === "DESCEND" ? "ArrowDownward" : "ArrowUpward" 
-  const Asset = Assets[iconType] 
+  const iconType = isSelected && sorting?.order === "DESCEND" ? "ArrowDownward" : "ArrowUpward"
+  const Asset = Assets[iconType]
 
   const onSorterClick = () => {
-    const newOrder = isSelected && sorting.order === "ASCEND" ? "DESCEND" : "ASCEND"
-    onChangeSorting({ columnKey: index, order: newOrder })
+    const newOrder = isSelected && sorting?.order === "ASCEND" ? "DESCEND" : "ASCEND"
+    onChangeSorting({ index: index, order: newOrder })
```

Comment:
Just `{ index, order: newOrder }` is enough, right?

Reply:
Yes, here it is. But `columns` now has a `dataIndex` property. Which can also be used to select a value from a data row.